src/js/classes/resizable.js

```diff
@@ -0,0 +1,503 @@
+          data.top = null;
+        }
+        if (this._axis === 'nw') {
+          data.top = this._position.top + (this._size.height - data.height);
+          data.left = this._position.left + (this._size.width - data.width);
+        }
+
+        return data;
+      };
+
+
+      /**
+       * Updates current position and size from boundaries data.
+       *
+       * @param data - boundaries data. 
+       */
+      ResizableFactory.prototype.updateSizeAndPosition = function(data) {
+        if (angular.isNumber(data.left)) {
+          this._position.left = data.left;
+        }
+        if (angular.isNumber(data.top)) {
+          this._position.top = data.top;
+        }
+        if (angular.isNumber(data.height)) {
+          this._size.height = data.height;
+        }
+        if (angular.isNumber(data.width)) {
+          this._size.width = data.width;
+        }
+      };
+
+      /**
+       * Calculates new virtual boundaries values depends on options and ratio.
+       */
+      ResizableFactory.prototype.updateVirtualBoundaries = function() {
+        var pMinWidth, pMaxWidth, pMinHeight, pMaxHeight, b,
+            o = this._options;
+
+        b = {
+          minWidth: angular.isNumber(o.minWidth) ? o.minWidth : 0,
+          maxWidth: angular.isNumber(o.maxWidth) ? o.maxWidth : Infinity,
+          minHeight: angular.isNumber(o.minHeight) ? o.minHeight : 0,
+          maxHeight: angular.isNumber(o.maxHeight) ? o.maxHeight : Infinity
+        };
+
+        if (this._ratio) {
+          pMinWidth = b.minHeight * this._ratio;
+          pMinHeight = b.minWidth / this._ratio;
+          pMaxWidth = b.maxHeight * this._ratio;
+          pMaxHeight = b.maxWidth / this._ratio;
+
+          if (pMinWidth > b.minWidth) {
+            b.minWidth = pMinWidth;
+          }
+          if (pMinHeight > b.minHeight) {
+            b.minHeight = pMinHeight;
+          }
+          if (pMaxWidth < b.maxWidth) {
+            b.maxWidth = pMaxWidth;
+          }
+          if (pMaxHeight < b.maxHeight) {
+            b.maxHeight = pMaxHeight;
+          }
+        }
+
+        this._vBoundaries = b;
+      };
+
+      /**
+       * Updates boundaries data to fit min/max sizes.
+       *
+       * @param data - boundaries data. 
+       */
+      ResizableFactory.prototype.respectSize = function(data) {
+        var o = this._vBoundaries,
+            a = this._axis,
+            ismaxw = angular.isNumber(data.width) && o.maxWidth && (o.maxWidth < data.width),
+            ismaxh = angular.isNumber(data.height) && o.maxHeight && (o.maxHeight < data.height),
+            isminw = angular.isNumber(data.width) && o.minWidth && (o.minWidth > data.width),
+            isminh = angular.isNumber(data.height) && o.minHeight && (o.minHeight > data.height),
+            dw = this._originalPosition.left + this._originalSize.width,
+            dh = this._position.top + this._size.height,
+            cw = /sw|nw|w/.test(a), ch = /nw|ne|n/.test(a);

```

Comment:
Variable names :-1: 

Why not use `options` and `axis`?
`ismaxw` not using any camelCase letter like `isMaxW`

Also can I suggest to use `isMaxWidth` to make it more readable?
(same applies for the other variables onward)

What does `dw` stands for? can you make a more extended name?